src/components/JobCard/JobCard.tsx

```diff
@@ -50,9 +77,28 @@ function JobCard({
     } catch (e) {
       console.log(e);
     }
-    setFilteredJobs(filteredJobs.filter((_, index) => index !== idx));
+    if (filteredJobs === null) {
+      return;
+    }
+    setFilteredJobs?.(filteredJobs.filter((_, index) => index !== idx));
   }
 
+  const toggleBookmark = async (val: boolean | undefined) => {
+    if (userObject !== null) {
+      if (userObject?.bookmarkedJobs?.includes(job.id)) {
+        const index = userObject?.bookmarkedJobs.indexOf(job.id);
```

Comment:
might recommend moving the logic on lines 89/90 to a helper function removeBookmarkedJob so it's easier to know what's being done here